### PR TITLE
feat: Add equipment

### LINKED ISSUE #1465: API equippment_contents
image

### PR DESCRIPTION
**Pull request by DuckySoLucky**
## Description 

Closes #1465 

## Example
image
imageimage

### CODE REVIEW

**Comment by DuckySoLucky** [comment]
## To-do 
- [x] ~~add `Bonus` like in armor (will finish it tomorrow)~~ Doesn't seem to work (Not sure why) [Click Here for more information](https://canary.discord.com/channels/738971489411399761/748625263893545042/1018222831185703042)

**Comment by dukio** [comment]
> ## To-do
> * [x]  ~add `Bonus` like in armor (will finish it tomorrow)~ Doesn't seem to work (Not sure why) [Click Here for more information](https://canary.discord.com/channels/738971489411399761/748625263893545042/1018222831185703042)

If you merge https://github.com/DuckySoLucky/SkyCrypt/pull/1 it should work and be all good 😄 

(i'll wait until merge for full review of this PR, but everything else should be good)

**Review by dukio — APPROVED** [review]
lgtm, tested

**Comment by MartinNemi03** [comment]
Seems like there is a conflict in the farming section

**Review by MartinNemi03 — APPROVED** [review]
while i would definitely consider merging armor and equipment categories in the future, this is a good first step in adding what was missing

lgtm, tested

**Comment by DuckySoLucky** [comment]
> Seems like there is a conflict in the farming section

Solved

**Comment by DuckySoLucky** [comment]
> while i would definitely consider merging armor and equipment categories in the future, this is a good first step in adding what was missing

I wanted to do that in first place, but people on discord server said to split it so 🤷‍♂️. If you want I can merge it

**Comment by MartinNemi03** [comment]
> > while i would definitely consider merging armor and equipment categories in the future, this is a good first step in adding what was missing
> 
> I wanted to do that in first place, but people on discord server said to split it so 🤷‍♂️. If you want I can merge it

If that's so, then we should leave it this way for now.

**Review by nstringham — CHANGES_REQUESTED** [review]
> > while i would definitely consider merging armor and equipment categories in the future, this is a good first step in adding what was missing
> 
> I wanted to do that in first place, but people on discord server said to split it so 🤷‍♂️. If you want I can merge it

We should definitely merge it now

if they are 2 separate tabs it makes an awful scrollbar appear on desktop
image

plus I really don't like having Wardrobe in the Equipment section

File: src/lib.js
Comment (nstringham) [inline src/lib.js:893]: what does `.filter((x) => x.rarity)` do?

File: src/lib.js
Comment (nstringham) [inline src/lib.js:1349]: do equipment sets even exist?

File: src/lib.js
Comment (DuckySoLucky) [reply src/lib.js:893]: Not really sure, the armor has it so I thought equipment needed it as well. Since equipment got sets and other stuff like armor

File: src/lib.js
Comment (DuckySoLucky) [reply src/lib.js:1349]: Yeah they do

**Comment by DuckySoLucky** [comment]
> > > while i would definitely consider merging armor and equipment categories in the future, this is a good first step in adding what was missing
> > 
> > 
> > I wanted to do that in first place, but people on discord server said to split it so 🤷‍♂️. If you want I can merge it
> 
> We should definitely merge it now
> 
> if they are 2 separate tabs it makes an awful scrollbar appear on desktop image
> 
> plus I really don't like having Wardrobe in the Equipment section

Okay, I will update that. Most likely on Wednesday since I'm busy this week :/

**Comment by DuckySoLucky** [comment]
> > > while i would definitely consider merging armor and equipment categories in the future, this is a good first step in adding what was missing
> > 
> > 
> > I wanted to do that in first place, but people on discord server said to split it so 🤷‍♂️. If you want I can merge it
> 
> We should definitely merge it now
> 
> if they are 2 separate tabs it makes an awful scrollbar appear on desktop image
> 
> plus I really don't like having Wardrobe in the Equipment section

image
image
Opinions on this? Image 1 or 2? Maybe use 2nd image but with combined equipment and armor bonus?

**Comment by nstringham** [comment]
> image image Opinions on this? Image 1 or 2? Maybe use 2nd image but with combined equipment and armor bonus?

Maybe like the first one but with a subheading between them that says equipment

**Comment by DuckySoLucky** [comment]
> > image image Opinions on this? Image 1 or 2? Maybe use 2nd image but with combined equipment and armor bonus?
> 
> Maybe like the first one but with a subheading between them that says equipment

image
Like this?

**Comment by nstringham** [comment]
> > Maybe like the first one but with a subheading between them that says equipment
> 
> image Like this?

I mean a little heading like the one for wardrobe

**Comment by DuckySoLucky** [comment]
image

File: public/resources/ts/calculate-player-stats.ts
Comment (nstringham) [inline public/resources/ts/calculate-player-stats.ts:56]: do we ever display this?

File: src/lib.js
Comment (nstringham) [inline src/lib.js]: is this ever displayed?

File: public/resources/ts/calculate-player-stats.ts
Comment (DuckySoLucky) [reply public/resources/ts/calculate-player-stats.ts:56]: image

File: src/lib.js
Comment (DuckySoLucky) [reply src/lib.js]: I don't think so, most likely will remove

File: public/resources/ts/calculate-player-stats.ts
Comment (nstringham) [reply public/resources/ts/calculate-player-stats.ts:56]: Cool

File: src/lib.js
Comment (nstringham) [reply src/lib.js]: Ok

**Comment by dukio** [comment]
Fixed couple things in https://github.com/DuckySoLucky/SkyCrypt/pull/3 and tested, this can be merged

File: src/lib.js
Comment (nstringham) [reply src/lib.js:893]: I saw there was some discussion with dukio about this but I didn't see the resolution. What did this do again?

File: src/lib.js
Comment (nstringham) [inline src/lib.js:1282]: Are we using this?

File: src/lib.js
Comment (DuckySoLucky) [reply src/lib.js:893]: > I saw there was some discussion with dukio about this but I didn't see the resolution. What did this do again?

It fixes equipment being sorted wrongly

File: src/lib.js
Comment (nstringham) [inline src/lib.js:1351]: It feels like we would be better off just filtering in the first place. I think I have seen this 3 times now

File: src/lib.js
Comment (dukio) [reply src/lib.js:893]: Basically we need to filter out null items otherwise it breaks the all items Map in stats defer and other issues too iirc

**Pull request merged**
